library/ssl_tls.c

```diff
@@ -6600,64 +6600,89 @@ int mbedtls_ssl_set_calc_verify_md(mbedtls_ssl_context *ssl, int md)
     return 0;
 }
 
-#if defined(MBEDTLS_MD_CAN_SHA256)
-int ssl_calc_verify_tls_sha256(const mbedtls_ssl_context *ssl,
-                               unsigned char *hash,
-                               size_t *hlen)
-{
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-    size_t hash_size;
+static int ssl_calc_verify_tls_psa(const mbedtls_ssl_context *ssl,
+                                   const psa_hash_operation_t *hs_op,
+                                   size_t buffer_size,
+                                   unsigned char *hash,
+                                   size_t *hlen)
+{
     psa_status_t status;
-    psa_hash_operation_t sha256_psa = psa_hash_operation_init();
+    psa_hash_operation_t copy = psa_hash_operation_init();
 
-    MBEDTLS_SSL_DEBUG_MSG(2, ("=> PSA calc verify sha256"));
-    status = psa_hash_clone(&ssl->handshake->fin_sha256_psa, &sha256_psa);
+#if !defined(MBEDTLS_DEBUG_C)
+    (void) ssl;
+#endif
+    MBEDTLS_SSL_DEBUG_MSG(2, ("=> PSA calc verify"));
```

Comment:
Ah, thanks for the suggestion! However it only works with MD, not with PSA, and since PSA is the future, I don't think we need to improve the case that uses MD.